zerocopy-derive/src/enum.rs

```diff
@@ -57,7 +26,7 @@ pub(crate) fn generate_tag_enum(repr: &EnumRepr, data: &DataEnum) -> TokenStream
     });
 
     quote! {
-        #[repr(#repr)]
+        #repr
```

Comment:
This new behavior implicates #1758. In particular, this now delegates to the `ToTokens` impl for `EnumRepr`, which includes `align` reprs.

cc @djkoloski

Reply:
IMO the right thing to do here is to strip out any align attribute and only keep transparent, c, or int. That will ensure that this PR doesn't change behavior.

Then, we should shortly afterwards follow up with https://github.com/google/zerocopy/issues/1758 and resolve any bugs.